### ISSUE TITLE
[BUG] PR #59499 changes behaviour of random.get_str() and breaks backward compatibility with no standard notice period

### ISSUE THREAD

**Issue by ggiesen** [saltstack/salt#61221]
**Description**
The changes in #59599 to address #59486 changes the behaviour of `random.get_str` to include punctuation by default. This broke https://github.com/salt-formulas/salt-formula-freeipa (see salt-formulas/salt-formula-freeipa#19) , and may break other's states as well. At a minimum `punctuation = False` should have been the default to have the function spit out the same character set as before, with the standard notice period of 2 releases before the behaviour change to allow the rest of the world to catch up (although I'm not convinced `punctuation = True` ever needs to be the default).

As a result I lost many hours trying to debug a formula that previously worked . I absolutely love Salt, and am very grateful to the contributor who submitted the improvements, but the most frustrating part of using the project is the continuous, unnecessary mass-breakage between major versions.

Documentation for the module is available here:

https://docs.saltproject.io/en/latest/ref/modules/all/salt.modules.mod_random.html

**Comment by waynew** [comment]
I don't know if it's worth changing that default value back if we do a point release - but we should put in a PR to update the changelog, at *minimum*. 

It might be worth changing `punctuation=False` and adding a notice that it will change to a wider default set in the future :shrug:, especially if we plan on a point release.

**Comment by anitakrueger** [comment]
I have experienced this as well now in all our states that use random.get_str to generate a password. Sometimes it works (if the punctuation isn't at the beginning of the string for example) and sometimes it doesn't. Luckily I made a macro for this a long time ago so only had to change one place in our code and set punctuation to `false`. However...now I need all my salt minions to be upgraded, because otherwise they don't recognize the parameter.

Long story short...punctuation should be false by default. Agree with @ggiesen completely!

**Comment by ggiesen** [comment]
Going to close this one as there's been no movement one way or the other
